sql/query/reindex.go

```diff
@@ -30,7 +30,8 @@ func (stmt ReIndexStmt) Run(ctx context.Context, tx *database.Transaction, args
 	if err == nil {
 		return res, t.ReIndex()
 	}
-	if err != database.ErrTableNotFound {
+	errReturned  := database.ErrTableNotFound(stmt.TableOrIndexName)
```

Comment:
In particular, this comparison should have been `errors.Is(err, database.ErrTableNotFound)`.